src/classes/OPP_SendAcknowledgmentBTN_CTRL.cls

```diff
@@ -51,60 +56,75 @@ public with sharing class OPP_SendAcknowledgmentBTN_CTRL {
     public Boolean redirect { get; set; }
 
     /*******************************************************************************************************
-    * @description the list of Opportunities to be acknowledged
+    * @description List of Opportunities that should receive email acknowledgements 
     */
-    private list<Opportunity> listOpp { 
+    @TestVisible
+    private List<Opportunity> oppsToAcknowledge { 
         get {
-            if (listOpp == null)
-                listOpp = new list<Opportunity>(); 
-            return listOpp;
+            if (oppsToAcknowledge == null) {
+                oppsToAcknowledge = [
+                    SELECT Acknowledgment_Status__c 
+                    FROM Opportunity
+                    WHERE Id IN :oppIds
+                ]; 
+            }
+            return oppsToAcknowledge;
         }
         set; 
     }
 
+    /*******************************************************************************************************
+    * @description Ids of Opportunities that should receive email acknowledgements
+    */
+    private Set<Id> oppIds = new Set<Id>();
+
     /*******************************************************************************************************
     * @description the list of Opportunities that were not acknowledged
     */
-    public list<Opportunity> listOppNotSent { 
-        get {
-            if (listOppNotSent == null)
-                listOppNotSent = new list<Opportunity>(); 
-            return listOppNotSent;
-        }
-        set; 
-    }
+    public List<Opportunity> listOppNotSent { get; set; }
```

Comment:
This one follows a Hungarian notation pattern, where the data type prefixes the var name. I suggest renaming to `oppsNotSent`.

Reply:
@force2b I've renamed it.  FYI - @RadSFDeveloper  reminded me that the name changes might require some existing dev orgs to blank out the visual force page (because of the dependency on the variable) and then redeploy the page.

Reply:
Thanks @patrick-yan-sf, though I'm not sure about the rest. Only a class var was renamed, not an actual Visualforce Page or Apex Class. No changes should be required by customers at all, and we definitely wouldn't want to require any. Am I missing something on this @RadSFDeveloper?

Reply:
@force2b 
I think @RadSFDeveloper meant this change would only impact our internal DEV orgs.  Customers would not be impacted.

Reply:
Interesting. I've never had to do that for any var name changes.

Reply:
@force2b , I thought internal Dev orgs are impacted if VF page controller's public vars/methods are renamed. If that is not the case, please ignore the comment. No custom org is affected.